▶Buttons pygame

Lakhankumawat opened the pull request:
<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.
Help us understand your motivation by explaining why you decided to make this change.
Happy Contributing!
-->

# Description

Please include a summary of the change and which issue is fixed. List any dependencies that are required for this change.

## Fixes #904



## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

- [x] Yes
- [ ] No

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] My documentation follows [`Template for README.md`](https://github.com/Lakhankumawat/Amazing-Python-Scripts/blob/Buttons-Pygame/Pygame-Buttons/README.md)
- [x] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works.

## Project Metadata

`` If there is no-file/nothing to fill the below fields with, then type: none ``

`` Example: `` If no requirements.txt needed/present, then type none in ``Requirments``.

Category:
- [ ] Calculators
- [ ] AI/ML
- [ ] Scrappers
- [ ] Social_Media
- [ ] PDF
- [ ] Image_Processing
- [ ] Video_Processing
- [x] Games
- [ ] Networking
- [ ] OS_Utilities
- [ ] Automation
- [ ] Cryptography
- [ ] Computer_Vision
- [x] Fun
- [ ] Others

Title: Pygame-Buttons

Folder: Pygame-Buttons

Requirements: requirements.txt

Script: main.py

Arguments: none

Contributor: Lakhan Kumawat

Description: How to make design and animate buttons in python.

Lakhankumawat commented:
### 🙇Apologies beforehand
> Although my code works fine 
- As when you will go through code you'll see i haven't Added comments in between code wherever needed so it is difficult to understood
- Not added  more spacing between codes to look it bit more nicerer
- For any changes i will respond lately cause i have my exams going on but i'll do respond
- I don't know how to control the flow of functions in another py file which we are using as module from a py file thast's the reason if you'll see in my main file i import module only when a button is clicked and not at first , i know this is not a good practice to do it but seems like i tried and this was the only option i had .

kaustubhgupta reviewed (changes requested):
For importing other file as module, to the following:
- Create an empty file called `__init__.py` in your project folder.
- Import your ExampleButtons on the top!

Explanation for this: `__init__.py` makes your directory act as a package. From package, you can import any module like we use external libraries. 

Also, The script didn't worked (error with color definition, possible solution: tuple definition of colors (255, 0, 0) for red and so on):

![image](https://user-images.githubusercontent.com/43691873/115346487-d93db680-a1cd-11eb-997c-7c966c8610b8.png)

Lakhankumawat commented:
okay i will try to fix them , infact just writing color name works in my spyder ide or maybe i m just using any module idk, i will make them in RGB format instead .
For importing modules i will learn it.

Lakhankumawat commented:
Please have a look once again , although that module on i will solve later .

kaustubhgupta reviewed (commented):
I am still not able to run your code:

![image](https://user-images.githubusercontent.com/43691873/115814685-2eb6d500-a413-11eb-98a3-b5fa17a2385a.png)

Lakhankumawat commented:
![ot](https://user-images.githubusercontent.com/55774240/115816547-a9cdba80-a416-11eb-94cc-de6b612eaf1f.png)

Works fine in my case , i will look at it

kaustubhgupta reviewed (commented):
I am still not able to run your code.

![image](https://user-images.githubusercontent.com/43691873/116033130-82792680-a67e-11eb-8c26-82be81ffb22d.png)

@santushtisharma10 @antrikshmisri can you confirm if this is only my error or you are also not able to run this?

antrikshmisri commented:
> I am still not able to run your code.
> 
> ![image](https://user-images.githubusercontent.com/43691873/116033130-82792680-a67e-11eb-8c26-82be81ffb22d.png)
> 
> @santushtisharma10 @antrikshmisri can you confirm if this is only my error or you are also not able to run this?

@kaustubhgupta The script is running fine
![Screenshot (208)](https://user-images.githubusercontent.com/54466356/116034039-2fa06e80-a680-11eb-9f7a-85ab74d6408d.png)

kaustubhgupta commented:
> > I am still not able to run your code.
> > ![image](https://user-images.githubusercontent.com/43691873/116033130-82792680-a67e-11eb-8c26-82be81ffb22d.png)
> > @santushtisharma10 @antrikshmisri can you confirm if this is only my error or you are also not able to run this?
> 
> @kaustubhgupta The script is running fine
> ![Screenshot (208)](https://user-images.githubusercontent.com/54466356/116034039-2fa06e80-a680-11eb-9f7a-85ab74d6408d.png)

okay, mine issue then, apologies @Lakhankumawat! @antrikshmisri you can review this PR

Lakhankumawat commented:
Make sure rest of all png buttons and others scripts attached are working fine too , they are working fine my side .

santushtisharma10 commented:
![image](https://user-images.githubusercontent.com/60578091/116352535-f2201a80-a812-11eb-95d4-63cd49736c38.png)
![image](https://user-images.githubusercontent.com/60578091/116352584-019f6380-a813-11eb-9428-7f0c4dc39b42.png)

@Lakhankumawat fix it

Lakhankumawat commented:
In pygame it's not possible to make a newline in a sentence so i tried but this is not fixable 
Also same goes for input cause i have to take input in single rect 

![STK-20210415-WA0055-6](https://user-images.githubusercontent.com/55774240/116491190-eede6a80-a8b6-11eb-91b2-06f98afc0537.png)

Lakhankumawat commented:
Reviewers please review it or close it ....

antrikshmisri commented:
> In pygame it's not possible to make a newline in a sentence so i tried but this is not fixable 
> Also same goes for input cause i have to take input in single rect 
> 
> 
> 
> 

You can do that by splitting the string to render at `\n`. Then to blit the text, run another loop with range from 0 - length of list obtained by splitting and offset the y position by the current value of loop variable

kaustubhgupta commented:
> > In pygame it's not possible to make a newline in a sentence so i tried but this is not fixable
> > Also same goes for input cause i have to take input in single rect
> 
> You can do that by splitting the string to render at `\n`. Then to blit the text, run another loop with range from 0 - length of list obtained by splitting and offset the y position by the current value of loop variable

@Lakhankumawat will this work?

Lakhankumawat commented:
> > > In pygame it's not possible to make a newline in a sentence so i tried but this is not fixable
> > > Also same goes for input cause i have to take input in single rect
> > 
> > 
> > You can do that by splitting the string to render at `\n`. Then to blit the text, run another loop with range from 0 - length of list obtained by splitting and offset the y position by the current value of loop variable
> 
> @Lakhankumawat will this work?

Sir i tried but i could not get the way @antrikshmisri  sir told me to do . 
I tried my best and if you want to close pr then your choice but i tried . Ok

kaustubhgupta commented:
> > > > In pygame it's not possible to make a newline in a sentence so i tried but this is not fixable
> > > > Also same goes for input cause i have to take input in single rect
> > > 
> > > 
> > > You can do that by splitting the string to render at `\n`. Then to blit the text, run another loop with range from 0 - length of list obtained by splitting and offset the y position by the current value of loop variable
> > 
> > 
> > @Lakhankumawat will this work?
> 
> Sir i tried but i could not get the way @antrikshmisri sir told me to do .
> I tried my best and if you want to close pr then your choice but i tried . Ok

No worries, I was just asking if the workaround worked or not. We will approve and merge this PR.

@antrikshmisri anything you would like to add?

antrikshmisri commented:
@kaustubhgupta @Lakhankumawat No problem, we can merge it as is

kaustubhgupta reviewed (changes requested):
@Lakhankumawat  Add the requirements.txt file and update the PR template with project metadata. Look at this PR for reference: https://github.com/avinashkranjan/Amazing-Python-Scripts/pull/1002

Lakhankumawat commented:
- Updated PR template with project metadata
- Added [Requirements.txt](https://github.com/Lakhankumawat/Amazing-Python-Scripts/blob/Buttons-Pygame/Pygame-Buttons/requirements.txt)

kaustubhgupta commented on Pygame-Buttons/requirements.txt:
```diff
@@ -0,0 +1,4 @@
+  | Requirement |  Command         | Documentation |
+--|-------------|------------------|---------------|
+1.| Pygame      |pip install pygame|  Pygame/PyPi  |
+--|-------------|------------------|---------------|
```
This not the way of defining the requirements file. See this requirement file: https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Amazon-Price-Alert/requirements.txt

Lakhankumawat replied:
Sorry , i didn't knew that

Lakhankumawat commented:
Changed requirements.txt as requested .
